### PR TITLE
ESQL: Add parsing for a LOOKUP command

### PR DESCRIPTION
**Pull request by nik9000**
This command will serve as a sort of "inline" enrich. This commit itself is mostly antlr generated code and paranoid tests that the new `LOOKUP` keyword doesn't clash with any variables named `lookup`.

I've also marked our ANTLR generated files as `linguist-generated` which causes them to be hidden by default in github's UI. You can still click a button to see them if you like. See https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github

### CODE REVIEW

**Review by costin — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
BWC failure is real - LOOKUP can't be used there. Easy enough.

File: x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4 [hunk 1]
```diff
@@ -268,3 +273,7 @@ enrichCommand
 enrichWithClause
     : (newName=qualifiedNamePattern ASSIGN)? enrichField=qualifiedNamePattern
     ;
+
+lookupCommand
+    : LOOKUP tableName=qualifiedNamePattern ON matchFields=qualifiedNamePatterns
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4:278]: I'm really not sure if this should use a qualified name pattern for the table name. IMHO the table name should be somewhat in line with `FROM`'s index name and `ENRICH`'s policy name. Qualified name patterns are usually used for field names, so we can do stuff like
```
KEEP `some-thing`.`something-else`.yet_something_else
```

`FROM` uses an `INDEX_UNQUOTED_IDENTIFIER`, `ENRICH` a similar `ENRICH_POLICY_NAME_BODY`.

Maybe we just align with `FROM` here and use `INDEX_UNQUOTED_IDENTIFIER` as well, unless table names need to be more general than that?

**Review by alex-spies — COMMENTED** [review]
Looks good; I think we could make the table name less general, though. (See below.)

File: x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4 (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4:278]: I'll give it a go.

**Review by alex-spies — APPROVED** [review]

File: .gitattributes [hunk 2]
```diff
@@ -3,3 +3,8 @@ CHANGELOG.asciidoc  merge=union
 # These files contain expected text output, and should not be changed on
 # Windows
 build-tools-internal/src/test/resources/org/elasticsearch/gradle/internal/release/*.asciidoc text eol=lf
+
+x-pack/plugin/esql/src/main/antlr/*.tokens linguist-generated=true
+x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/*.interp linguist-generated=true
+x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlBaseLexer*.java linguist-generated=true
+x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlBaseParser*.java linguist-generated=true
```
[the remark this answers is not in the document]
Comment (ChrisHegarty) [reply .gitattributes:10]: ha! interesting. what does the IDE do, if anything, with this ?

[the remark this answers is not in the document]
Comment (nik9000) [reply .gitattributes:10]: So far as I can tell, nothing.

**Pull request merged**